src/test/java/newbank/server/TestNewBank.java

```diff
@@ -20,21 +21,34 @@ public static void setupBank() throws DuplicateCustomerException, PasswordInvali
   }
 
   @Test
-  public void canRegisterNewCustomer() throws DuplicateCustomerException, PasswordInvalidException {
+  public void canRegisterNewCustomer()
+      throws DuplicateCustomerException, PasswordInvalidException, UsernameInvalidException {
     assertThat(bank.getCustomer("customer2").isPresent(), equalTo(false));
     bank.addCustomer("customer2", "123456");
     assertThat(bank.getCustomer("customer2").isPresent(), equalTo(true));
   }
 
   @Test(expected = DuplicateCustomerException.class)
-  public void throwsDuplicateCustomerWhenAccountAlreadyExists()
-      throws DuplicateCustomerException, PasswordInvalidException {
+  public void throwsWhenAddingDuplicateCustomer()
+      throws DuplicateCustomerException, PasswordInvalidException, UsernameInvalidException {
     bank.addCustomer("customer3", "123456");
     bank.addCustomer("customer3", "123456");
   }
 
+  @Test(expected = UsernameInvalidException.class)
+  public void throwsWhenRegisteringWithInvalidName()
```

Comment:
Could more exhaustively test usernames with a parameterized test. At a glance, it seems JUnit4 doesn't let you parameterize just one test in a class, but JUnit 5 does. So leave it if it's too much of a faff!